src/modules/roc_fec/reader.cpp

```diff
@@ -82,18 +82,17 @@ packet::PacketPtr Reader::read_() {
                 source_ = pp->rtp()->source;
                 has_source_ = true;
             }
-            cur_block_sn_ = pp->rtp()->seqnum;
-            skip_repair_packets_();
+            cur_sbn_ = pp->fec()->source_block_number;
+            drop_outdated_repair_packets_();
```

Comment:
Renaming skip_repair_packets_ to drop_outdated_repair_packets_ is not quite correct. 

Here we handle the (typical) situation when we are starting in the middle of a block. In this case we wait for the first source packet and then drop all repair packet from previous blocks. So originally these packets were not outdated, but we make them outdated by waiting for the source packet from the next block.

I would name this kind of drop somehow else to avoid confusion with really outdated packets. Maybe "dropping preceding repair packets".

Reply:
Well, I agree that they are not outdated. Probably it will be better to name it:

* skip_prev_block()
* skip_packets_from_prev_block()
* skip_queued_packets()

I would prefer to use `skip_queued_packets`. Probably it is also not a good name, but it
tells the code reader the truth.

Reply:
Well, the logic is good, but the naming is not quite true because this method does not simply drop all queued packets unconditionally. A correct name would be something like "drop_repair_packets_from_prev_blocks_()`.

We should also fix the log message:

```
        roc_log(LogTrace, "fec reader: dropping outdated repair packet,"
                          " decoding not started: cur_sbn=%lu pkt_sbn=%lu",
                          (unsigned long)cur_sbn_,
                          (unsigned long)fec->source_block_number);
```

Reply:
Fixed